compiler/src/iree/compiler/InputConversion/Common/AutoInputConversionPipeline.cpp

```diff
@@ -99,12 +104,50 @@ static void populateFeatures(Operation *op, const Dialect *chloDialect,
 
 void AutoInputConversionPipelinePass::runOnOperation() {
   ModuleOp module = getOperation();
-  MLIRContext *ctxt = &getContext();
+  MLIRContext *context = &getContext();
+
+  // Check if any plugin-provided pipeline extensions can convert dialects in
+  // the module first.
+  if (pipelineExtensions) {
+    llvm::StringSet<> detectedTypeMnemonics;
+    pipelineExtensions->populateDetectedCustomInputConversionTypes(
+        module, detectedTypeMnemonics);
+
+    if (detectedTypeMnemonics.getNumItems() >= 2) {
+      // TODO(scotttodd): handle multiple typeMnemonics (use all?)
+      module.emitError("not yet implemented mixture of input types");
```

Comment:
This will be a common error message that users encounter, if history is a guide. Perhaps include some advice (i.e. set a specific --iree-input-type= or audit the code to understand why dialects are mixed)?

Reply:
Thanks, updated. Sample error message:

`error: mixture of input types not yet implemented, set '--iree-input-type=[type]' explicitly instead of using 'auto' or audit the input program to understand why dialects are mixed (detected: 'foo' 'torch')`